tests/foreman/api/test_permission.py

```diff
@@ -248,16 +260,35 @@ def test_positive_check_create(self):
             "create_*" role.
 
         :CaseImportance: Critical
+
+        :BZ: 1464137
         """
-        for entity_cls in (entities.Architecture, entities.Domain):
+        for entity_cls in (
+                entities.Architecture,
+                entities.Domain,
+                entities.ActivationKey):
             with self.subTest(entity_cls):
                 with self.assertRaises(HTTPError):
                     entity_cls(self.cfg).create()
                 self.give_user_permission(
                     _permission_name(entity_cls, 'create')
                 )
-                entity_id = entity_cls(self.cfg).create_json()['id']
-                entity_cls(id=entity_id).read()  # As admin user.
+                entity = entity_cls(self.cfg)
+                entity_fields = entity.get_fields()
+                if 'organization' in entity_fields:
+                    if isinstance(
+                            entity_fields['organization'], OneToManyField):
+                        entity.organization = [self.org]
+                    else:
+                        entity.organization = self.org
+                if 'location' in entity_fields:
+                    # Currently entities with both org and loc are affected by
+                    # bug. Skip to the next entity
+                    if bz_bug_is_open('1464137'):
+                        continue
+                    entity.location = [self.loc]
```

Comment:
Technically it's not an entity with both fields but entity which has location, according to your `'location' in entity_fields` check. Also you're not covering the case of OneToOneField for location